packages/yoastseo/src/languageProcessing/researches/findKeyphraseInSEOTitle.js

```diff
@@ -102,37 +238,18 @@ const findKeyphraseInSEOTitle = function( paper, researcher ) {
 	}
 
 	// Check if the exact match of the keyphrase is found in the SEO title.
+	const prefixedFunctionWordsRegex = researcher.getConfig( "prefixedFunctionWordsRegex" );
 	const keywordMatched = wordMatch( title, keyword, locale, false );
 
-	if ( keywordMatched.count > 0 ) {
+	if ( keywordMatched.count > 0 && ! prefixedFunctionWordsRegex ) {
 		result.exactMatchFound = true;
 		result.allWordsFound = true;
 		result.position = adjustPosition( title, keywordMatched.position );
 
 		return result;
 	}
-	// If an exact match was not found, check the pre-sanitized version of the keyphrase for matching.
-	// This makes matching of keyphrases containing punctuation (e.g. ".rar") possible.
-	if ( ! keywordMatched ) {
-		const keywordMatchedBeforeSanitizing = keyword;
-		if ( keywordMatchedBeforeSanitizing.count > 0 ) {
-			result.exactMatchFound = true;
-			result.allWordsFound = true;
-			result.position = adjustPosition( title, keywordMatchedBeforeSanitizing.position );
-		}
-		return result;
-	}
```

Comment:
I removed this step since this step is actually never reached and executed.
After the removal, all the unit tests still pass and the unit tests specifically for `.rar` also still pass